### PR TITLE
build: golangci-lint: update go version configuration

### PR DESCRIPTION
**Pull request by mhofstetter**
**linters-settings.staticcheck.go**

Currently, the Go version for the staticcheck linter is configured via the property `linters-settings.staticcheck.go`. This has been deprecated in favor of the global config property `run.go` and results in the following warning.

```
❯ make lint
golangci-lint run
WARN [config_reader] The configuration option `linters.staticcheck.go` is deprecated, please use global `run.go`.
```

(or CI: https://github.com/cilium/cilium/actions/runs/8832188630/job/24249007710)

In addition, the new property `run.go` defaults to the Go version defined in the `go.mod` file.

Therefore, this commit removes the deprecated property in the golangci-lint config completely and depends on the fallback mechanism (`go.mod`).

**linters-settings.unused.go**

According to the golangci-lint documentation, the cofig property
`linters-settings.unused.go` doesn't exist (and AFAIU never has - according to some investigation)

https://golangci-lint.run/usage/linters/#unused

Hence, let's remove it.

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -42,17 +42,18 @@ run:
   #  - ".*\\.my\\.go$"
   #  - lib/bad.go
 
+  # Define the Go version limit.
+  # Mainly related to generics support since go1.18.
+  # Default: use Go version from the go.mod file, fallback on the env var `GOVERSION`, fallback on 1.17
+  go: "1.20"
```
Comment (mhofstetter) [inline .golangci.yaml]: follow-up question: should we remove this config completely and rely on the functionality that the version is taken from the `go.mod` file?

Comment (mhofstetter) [reply .golangci.yaml]: cc @tklauser

Comment (nebril) [reply .golangci.yaml]: I would delete it and see what happens.

**Review by nebril — APPROVED** [review]

File: .golangci.yaml (hunk 1 above)
Comment (tklauser) [reply .golangci.yaml]: Yup, agree that we should try to delete it.

Comment (mhofstetter) [reply .golangci.yaml]: Thanks :rocket: i removed it completely.

**Review by tklauser — APPROVED** [review]

**Pull request merged**